src/table/io/FitsSchemaInputMapper.cc

```diff
@@ -0,0 +1,619 @@
+        } else if (key->compare(0, 5, "TUNIT") == 0) {
+            int column = boost::lexical_cast<int>(key->substr(5)) - 1;
+            auto iter = _impl->byColumn().lower_bound(column);
+            if (iter == _impl->byColumn().end() || iter->column != column) {
+                iter = _impl->byColumn().insert(iter, FitsSchemaItem(column, -1));
+            }
+            _impl->byColumn().modify(iter, Impl::SetTUNIT(metadata.get<std::string>(*key)));
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TCCLS") == 0) {
+            int column = boost::lexical_cast<int>(key->substr(5)) - 1;
+            auto iter = _impl->byColumn().lower_bound(column);
+            if (iter == _impl->byColumn().end() || iter->column != column) {
+                iter = _impl->byColumn().insert(iter, FitsSchemaItem(column, -1));
+            }
+            _impl->byColumn().modify(iter, Impl::SetTCCLS(metadata.get<std::string>(*key)));
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TFORM") == 0) {
+            int column = boost::lexical_cast<int>(key->substr(5)) - 1;
+            auto iter = _impl->byColumn().lower_bound(column);
+            if (iter == _impl->byColumn().end() || iter->column != column) {
+                iter = _impl->byColumn().insert(iter, FitsSchemaItem(column, -1));
+            }
+            _impl->byColumn().modify(iter, Impl::SetTFORM(metadata.get<std::string>(*key)));
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TZERO") == 0) {
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TSCAL") == 0) {
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TNULL") == 0) {
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        } else if (key->compare(0, 5, "TDISP") == 0) {
+            if (stripMetadata) {
+                metadata.remove(*key);
+            }
+        }
+    }
+
+    // Find the column used to store flags, and setup the flag-handling data members from it.
+    _impl->flagColumn = metadata.get("FLAGCOL", 0);
+    if (_impl->flagColumn > 0) {
+        if (stripMetadata) {
+            metadata.remove("FLAGCOL");
+        }
+        --_impl->flagColumn; // switch from 1-indexed to 0-indexed
+        auto iter = _impl->byColumn().find(_impl->flagColumn);
+        if (iter == _impl->byColumn().end()) {
+            throw LSST_EXCEPT(
+                afw::fits::FitsError,
+                (boost::format("Column for flag data not found; FLAGCOL=%d") % _impl->flagColumn).str()
+            );
+        }
+        static boost::regex const regex("(\\d+)?X\\(?(\\d)*\\)?", boost::regex::perl);
```

Comment:
(Although I do note that this is not new code but rather being moved in from elsewhere, so it's fair enough to leave it as it is.)